runtime/interpreter/value.go

```diff
@@ -19586,8 +19624,32 @@ func (s SomeStorable) HasPointer() bool {
 	}
 }
 
+func getSomeStorableEncodedPrefixSize(nestedLevels uint64) uint32 {
+	if nestedLevels == 1 {
+		return cborTagSize
+	}
+	return cborTagSize + someStorableWithMultipleNestedlevelsArraySize + getUintCBORSize(nestedLevels)
+}
+
 func (s SomeStorable) ByteSize() uint32 {
-	return cborTagSize + s.Storable.ByteSize()
+	innermostStorable, nestedLevels := s.getInnermostStorable()
+	return getSomeStorableEncodedPrefixSize(nestedLevels) + innermostStorable.ByteSize()
+}
+
+// getInnermostStorable returns innermost storable of SomeStorable and nested levels.
+// For example, SomeStorable{true} has innermost storable true, and nested levels 1.
+func (s SomeStorable) getInnermostStorable() (atree.Storable, uint64) {
```

Comment:
Same as above: Maybe add another example and clarify the function naming, it could be misinterpreted.